### PR TITLE
Fix startup of docs container

### LINKED ISSUE #1185: [Unified Fides] Issues bringing up docs container
### Bug Description

Running the server with `nox -s dev -- mongo postgres` tries to bring up the docs container and fails with:
`ModuleNotFoundError: No module named 'fides'`

```
fides-fides-1              | 2022-10-03 18:33:37.016 [INFO] (main:log_request:260): Request received | {'method': 'GET', 'status_code': 200, 'handler_time': '726.729ms', 'path': '/health'}
fides-docs-1               | WARNING  -  Config value: 'dev_addr'. Warning: The use of the IP address '0.0.0.0' suggests a production environment or the use of a proxy to connect to the MkDocs server. However, the MkDocs' server is intended for local development purposes only. Please use a third party production-ready server instead.
fides-docs-1               | INFO     -  Building documentation...
fides-docs-1               | INFO     -  Cleaning site directory
fides-docs-1               | INFO     -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
fides-docs-1               |   - getting_started.md
fides-docs-1               |   - development/jetbrains_debugging.md
fides-docs-1               |   - development/release_checklist.md
fides-docs-1               |   - development/update_erd_diagram.md
fides-docs-1               |   - development/postman/using_postman.md
fides-docs-1               |   - guides/connection_types.md
fides-docs-1               |   - guides/creating_users.md
fides-docs-1               |   - guides/manual_webhooks.md
fides-docs-1               |   - postman/using_postman.md
fides-docs-1               |   - tutorial/index.md
fides-docs-1               |   - tutorial/add.md
fides-docs-1               |   - tutorial/dataset.md
fides-docs-1               |   - tutorial/google.md
fides-docs-1               |   - tutorial/pass.md
fides-docs-1               |   - tutorial/policy.md
fides-docs-1               |   - tutorial/system.md
fides-docs-1               |   - ui/deployment.md
fides-docs-1               |   - ui/local_dev.md
fides-docs-1               | PDF export is disabled (set environment variable PROD_PUBLISH to 1 to enable)
fides-docs-1               | ERROR    -  Error reading page 'cli.md': No module named 'fides'
fides-docs-1               | Traceback (most recent call last):
fides-docs-1               |   File "/usr/local/bin/mkdocs", line 8, in <module>
fides-docs-1               |     sys.exit(cli())
fides-docs-1               |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
fides-docs-1               |     return self.main(*args, **kwargs)
fides-docs-1               |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1055, in main
fides-docs-1               |     rv = self.invoke(ctx)
fides-docs-1               |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
fides-docs-1               |     return _process_result(sub_ctx.command.invoke(sub_ctx))
fides-docs-1               |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
fides-docs-1               |     return ctx.invoke(self.callback, **ctx.params)
fides-docs-1               |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 760, in invoke
fides-docs-1               |     return __callback(*args, **kwargs)
fides-docs-1               |   File "/usr/local/lib/python3.9/site-packages/mkdocs/__main__.py", line 234, in serve_command
fides-docs-1               |     serve.serve(dev_addr=dev_addr, livereload=livereload, watch=watch, **kwargs)
fides-docs-1               |   File "/usr/local/lib/python3.9/site-packages/mkdocs/commands/serve.py", line 82, in serve
fides-docs-1               |     builder(config)
fides-docs-1               |   File "/usr/local/lib/python3.9/site-packages/mkdocs/commands/serve.py", line 75, in builder
fides-docs-1               |     build(config, live_server=live_server, dirty=dirty)
fides-docs-1               |   File "/usr/local/lib/python3.9/site-packages/mkdocs/commands/build.py", line 308, in build
fides-docs-1               |     _populate_page(file.page, config, files, dirty)
fides-docs-1               |   File "/usr/local/lib/python3.9/site-packages/mkdocs/commands/build.py", line 181, in _populate_page
fides-docs-1               |     page.render(config, files)
fides-docs-1               |   File "/usr/local/lib/python3.9/site-packages/mkdocs/structure/pages.py", line 270, in render
fides-docs-1               |     self.content = md.convert(self.markdown)
fides-docs-1               |   File "/usr/local/lib/python3.9/site-packages/markdown/core.py", line 261, in convert
fides-docs-1               |     self.lines = prep.run(self.lines)
fides-docs-1               |   File "/usr/local/lib/python3.9/site-packages/mkdocs_click/_extension.py", line 48, in run
fides-docs-1               |     return list(
fides-docs-1               |   File "/usr/local/lib/python3.9/site-packages/mkdocs_click/_processing.py", line 35, in replace_blocks
fides-docs-1               |     yield from replace(**options)
fides-docs-1               |   File "/usr/local/lib/python3.9/site-packages/mkdocs_click/_extension.py", line 52, in <lambda>
fides-docs-1               |     replace=lambda **options: replace_command_docs(has_attr_list=self._has_attr_list, **options),
fides-docs-1               |   File "/usr/local/lib/python3.9/site-packages/mkdocs_click/_extension.py", line 28, in replace_command_docs
fides-docs-1               |     command_obj = load_command(module, command)
fides-docs-1               |   File "/usr/local/lib/python3.9/site-packages/mkdocs_click/_loader.py", line 15, in load_command
fides-docs-1               |     command = _load_obj(module, attribute)
fides-docs-1               |   File "/usr/local/lib/python3.9/site-packages/mkdocs_click/_loader.py", line 25, in _load_obj
fides-docs-1               |     mod = importlib.import_module(module)
fides-docs-1               |   File "/usr/local/lib/python3.9/importlib/__init__.py", line 127, in import_module
fides-docs-1               |     return _bootstrap._gcd_import(name[level:], package, level)
fides-docs-1               |   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
fides-docs-1               |   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
fides-docs-1               |   File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
fides-docs-1               |   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
fides-docs-1               |   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
fides-fides-1              | 2022-10-03 18:33:38.382 [INFO] (main:log_request:260): Request received | {'method': 'GET', 'status_code': 200, 'handler_time': '246.369ms', 'path': '/health'}
fides-docs-1               |   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
fides-docs-1               |   File "<frozen importlib._bootstrap>", line 984, in _find_and_load_unlocked
fides-docs-1               | ModuleNotFoundError: No module named 'fides'
```

#### Steps to Reproduce

nox -s dev -- mongo postgres

### Expected behavior

_A description of what you expected to happen._

### Screenshots

_If applicable, add screenshots to help explain your problem._

### Environment

* Version:
* OS:
* Python Version:
* Docker Version:

### Additional context

_Add any other context about the problem here._

### PR DESCRIPTION
**Pull request by sanders41**
Closes #1185

### Code Changes

* [x] Updated the docs `Dockerfile` to build in stages. The first stage builds everything and the second stage just copies in what is needed for the docs.
* [x] Updated the `nox` commands to build the doc container from the root directory. 

### Steps to Confirm

* [ ] Run `nox -s docs_build` and see no errors
* [ ] Run `nox -s docs_serve` and the docs container should start and docs should be viewable at `http:localhost:8000`
* [ ] Run docker-compose up docs and the docs container should start and docs should be viewable at `http:localhost:8000`

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation Updated:
  * [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  * [ ] documentation issue created (tag docs-team to complete issue separately)
* [x] Issue Requirements are Met
* [x] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`

### Description Of Changes

_Write some things here about the changes and any potential caveats_

### CODE REVIEW

File: noxfiles/docs_nox.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def docs_serve(session: nox.Session) -> None:
         "docs",
         "/bin/bash",
         "-c",
-        "pip install -e /fides && mkdocs serve --dev-addr=0.0.0.0:8000",
```
Comment (adamsachs) [inline noxfiles/docs_nox.py:36]: ‼️

**Review by adamsachs — APPROVED** [review]
this looks great, the multi-stage build works nicely here to get what we need without bloating the docs image too much.

my only question is whether we want to call out a bit more clearly (maybe in the PR description?) that we're effectively putting the `docs` subdir back into the base `fides` image build now, just to make sure that change can be traced back if we want to revisit it?

**Comment by sanders41** [comment]
Closing in favor of #1271

**Comment by sanders41** [comment]
Closing in favor of #1271

**Pull request closed without merging**